File: go/ecmr.go
package zymeup

type EcmrClient struct {
	client *Client
}

func NewEcmrClient(c *Client) *EcmrClient {
	return &EcmrClient{client: c}
}

func (c *EcmrClient) List(params map[string]interface{}) ([]map[string]interface{}, error) {
	path := "/api/v1/shipment/ecmr/list" + buildQueryString(params)
	resp, err := c.client.doRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int `json:"code"`
		Data struct {
			Items []map[string]interface{} `json:"items"`
			Total int                      `json:"total"`
		} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data.Items, nil
}

func (c *EcmrClient) Get(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("GET", "/api/v1/shipment/ecmr/"+id, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *EcmrClient) Create(data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/shipment/ecmr/create", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *EcmrClient) GenerateFromOrder(orderID string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/shipment/ecmr/generate-from-order", map[string]interface{}{
		"order_id": orderID,
	})
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *EcmrClient) Sign(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/shipment/ecmr/"+id+"/sign", nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *EcmrClient) Pdf(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/shipment/ecmr/"+id+"/pdf", nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

File: python/zymeup_sdk/ecmr.py
"""Zymeup Python SDK - ECMR Client"""
from typing import Optional, Dict, Any, List


class EcmrClient:
    """ECMR (European Consignment Note) client"""

    def __init__(self, client):
        self._client = client

    def list(self, page: int = 1, page_size: int = 25) -> Dict[str, Any]:
        """List ECMR records"""
        params = {"page": page, "page_size": page_size}
        return self._client.request("GET", "/api/v1/shipment/ecmr/list", params=params)

    def get(self, ecmr_id: str) -> Dict[str, Any]:
        """Get ECMR detail"""
        return self._client.request("GET", f"/api/v1/shipment/ecmr/{ecmr_id}")

    def create(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """Create ECMR"""
        return self._client.request("POST", "/api/v1/shipment/ecmr/create", json=data)

    def generate_from_order(self, order_id: str) -> Dict[str, Any]:
        """Generate ECMR from order"""
        return self._client.request("POST", "/api/v1/shipment/ecmr/generate-from-order", json={"order_id": order_id})

    def sign(self, ecmr_id: str) -> Dict[str, Any]:
        """Generate signing URL for ECMR"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/sign")

    def update(self, ecmr_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        """Update ECMR"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/update", json=data)

    def cancel(self, ecmr_id: str) -> Dict[str, Any]:
        """Cancel ECMR"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/cancel")

    def validate(self, ecmr_id: str) -> Dict[str, Any]:
        """Validate ECMR"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/validate")

    def submit_to_authority(self, ecmr_id: str) -> Dict[str, Any]:
        """Submit ECMR to authority"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/submit-to-authority")

    def pdf(self, ecmr_id: str) -> Dict[str, Any]:
        """Generate PDF for ECMR"""
        return self._client.request("POST", f"/api/v1/shipment/ecmr/{ecmr_id}/pdf")
